crates/oxc_linter/src/rules/unicorn/consistent_existence_index_check.rs

```diff
@@ -0,0 +1,496 @@
+use oxc_ast::{
+    ast::{BinaryOperator, Expression, UnaryOperator, VariableDeclarationKind},
+    AstKind,
+};
+
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_semantic::AstNode;
+use oxc_span::{GetSpan, Span};
+
+use crate::{context::LintContext, rule::Rule};
+
+#[derive(Debug, Default, Clone)]
+pub struct ConsistentExistenceIndexCheck;
+
+fn consistent_existence_index_check_diagnostic(label: String, span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(label).with_label(span)
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Enforce consistent style for element existence checks with `indexOf()`, `lastIndexOf()`, `findIndex()`, and `findLastIndex()`
+    ///
+    /// ### Why is this bad?
+    ///
+    /// This rule is only meant to enforce a specific style and make comparisons more clear.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    ///
+    /// ```javascript
+    /// const index = foo.indexOf('bar');
+    /// if (index < 0) {}
+    /// ```
+    ///
+    /// ``` javascript
+    /// const index = foo.indexOf('bar');
+    /// if (index >= 0) {}
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    ///
+    /// ```javascript
+    /// const index = foo.indexOf('bar');
+    /// if (index === -1) {}
+    /// ```
+    ///
+    /// ``` javascript
+    /// const index = foo.indexOf('bar');
+    /// if (index !== -1) {}
+    /// ```
+    ConsistentExistenceIndexCheck,
+    restriction,
+    fix,
+);
+
+const METHOD_NAMES: [&str; 4] = ["indexOf", "lastIndexOf", "findIndex", "findLastIndex"];
+
+impl Rule for ConsistentExistenceIndexCheck {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::BinaryExpression(binary_expression) = node.kind() {
```

Comment:
prefer `let AstKind::BinaryExpression(binary_expression) = node.kind() else {return; };` please, it helps avoid overly nested blocks